src/Beacon.jsx

```diff
@@ -242,6 +247,50 @@ export class Beacon extends Component {
     }
   }
 
+  loadHash(persistent) {
+    const { tooltipText } = this.props;
+    const text = typeof(tooltipText) === 'string' ? tooltipText : ReactDOMServer.renderToStaticMarkup(tooltipText);
+    const hash = persistent === true ? sha1(text) : persistent;
+
+    const transaction = this.state.database.transaction(['beacons']);
+    const objectStore = transaction.objectStore('beacons');
+    objectStore.get(hash).onsuccess = (event) => {
+      if (!event.target.result) {
+        if (!this.state.hash) {
+          this.setState({ hash });
+        }
+      } else {
+        if (this.state.hash) {
+          this.setState({ hash: null });
+        }
+      }
+    };
+  }
+
+  checkHash() {
```

Comment:
`checkHash` is called in `componentWillUpdate` yet you are calling `setState` inside, I know that now it's asynchronous so it shouldn't matter, but what if you changed the implementation later to something synchronous? The real point is, should it really be used in `componentWillUpdate` ?